osx/jhbuildrc-custom

```diff
@@ -101,14 +101,14 @@ module_autogenargs['tango-icon-theme']      ='--disable-nls --disable-icon-frami
 module_autogenargs['harfbuzz']      = '--disable-gtk-doc --disable-gtk-doc-html --disable-gtk-doc-pdf --with-coretext --without-icu '# + autogenargs
 module_autogenargs['root-harfbuzz'] = '--disable-gtk-doc --disable-gtk-doc-html --disable-gtk-doc-pdf --with-coretext --without-icu '# + autogenargs
 module_autogenargs['fontconfig']     = '--with-cache-dir="~/Library/Caches/org.gnome.meld.fontconfig" --with-default-fonts=/System/Library/Fonts --with-add-fonts=/Library/Fonts,/Network/Library/Fonts,/System/Library/Fonts --disable-cache-build --disable-docs --disable-dependency-tracking'
-#module_autogenargs['python3']  = '--enable-optimizations --disable-ipv6 --with-ensurepip=yes --with-system-expat --without-readline --with-pkg-config --disable-test-modules ' 
+module_autogenargs['python3']  = '--enable-optimizations --disable-ipv6 --with-ensurepip=yes --with-system-expat --without-readline --with-pkg-config --disable-test-modules --with-openssl=/opt/homebrew/opt/openssl@3/'
```

Comment:
This one will tie us to homebrew being installed at run-time. While the adjustment for paths during compile time are fine, we can't assume that homebrew is installed. It's better not to build support for SSL in python altogether since Meld won't need it.

Reply:
Right, but without this line, Python will not have SSL support (it only looks in `/usr`... by default). Without SSL support, installation fails later in `build_app.sh`:

```
*** using recipe: setuptools *** {'expected_missing_imports': {'pkg_resources.extern.packaging._manylinux', 'pkg_resources.extern.jaraco.context', 'pkg_resources.extern.importlib_resources._itertools', 'pkg_resources.extern.jaraco', 'pkg_resources.extern.importlib_resources.abc', 'pkg_resources.extern.packaging.specifiers', 'pkg_resources.extern.importlib_resources', 'pkg_resources.extern.platformdirs', 'pkg_resources.extern.jaraco.functools', 'pkg_resources.extern.platformdirs.unix', 'pkg_resources.extern.packaging.markers', 'pkg_resources.extern.platformdirs.version', 'pkg_resources.extern.packaging._elffile', 'pkg_resources.extern.packaging.metadata', 'pkg_resources.extern.more_itertools.recipes', 'pkg_resources.extern.packaging', 'pkg_resources.extern.more_itertools', 'pkg_resources.extern.typing_extensions', 'pkg_resources.extern.importlib_resources._legacy', 'pkg_resources.extern.more_itertools.more', 'pkg_resources.extern.packaging._structures', 'pkg_resources.extern.zipp', 'pkg_resources.extern.importlib_resources._compat', 'pkg_resources.extern.packaging.tags', 'pkg_resources.extern.platformdirs.__main__', 'pkg_resources.extern.packaging.utils', 'pkg_resources.extern.platformdirs.macos', 'pkg_resources.extern.packaging.requirements', 'pkg_resources.extern.jaraco.text', '__builtin__', 'pkg_resources.extern.platformdirs.android', 'pkg_resources.extern.importlib_resources.readers', 'pkg_resources.extern.packaging._musllinux', 'pkg_resources.extern.packaging._parser', 'pkg_resources.extern.platformdirs.windows', 'pkg_resources.extern.importlib_resources._adapters', 'pkg_resources.extern.platformdirs.api', 'pkg_resources.extern.importlib_resources.simple', '__main__.__requires__', 'pkg_resources.extern.packaging.version', 'pkg_resources.extern.packaging._tokenizer', 'pkg_resources.extern.importlib_resources._common'}, 'prescripts': [<_io.StringIO object at 0x1079369e0>]}
--- Skipping recipe PIL ---
--- Skipping recipe autopackages ---
--- Skipping recipe black ---
--- Skipping recipe gcloud ---
--- Skipping recipe lxml ---
--- Skipping recipe matplotlib ---
--- Skipping recipe opencv ---
--- Skipping recipe pandas ---
*** using recipe: platformdirs *** {'includes': ['platformdirs.macos']}
--- Skipping recipe PIL ---
--- Skipping recipe autopackages ---
--- Skipping recipe black ---
--- Skipping recipe gcloud ---
--- Skipping recipe lxml ---
--- Skipping recipe matplotlib ---
--- Skipping recipe opencv ---
--- Skipping recipe pandas ---
--- Skipping recipe pydantic ---
--- Skipping recipe pyenchant ---
--- Skipping recipe pygame ---
--- Skipping recipe pylsp ---
--- Skipping recipe pyopengl ---
--- Skipping recipe pyside ---
--- Skipping recipe pyside2 ---
--- Skipping recipe pyside6 ---
--- Skipping recipe qt5 ---
--- Skipping recipe qt6 ---
--- Skipping recipe rtree ---
--- Skipping recipe shiboken2 ---
--- Skipping recipe shiboken6 ---
--- Skipping recipe sip ---
--- Skipping recipe six ---
--- Skipping recipe sphinx ---
--- Skipping recipe sqlalchemy ---
Traceback (most recent call last):
  File "/Users/mdiener/Work/meld/setup_py2app.py", line 82, in <module>
    setup(
  File "/Users/mdiener/gtk/inst/lib/python3.11/site-packages/setuptools/__init__.py", line 103, in setup
    return distutils.core.setup(**attrs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/mdiener/gtk/inst/lib/python3.11/site-packages/setuptools/_distutils/core.py", line 185, in setup
    return run_commands(dist)
           ^^^^^^^^^^^^^^^^^^
  File "/Users/mdiener/gtk/inst/lib/python3.11/site-packages/setuptools/_distutils/core.py", line 201, in run_commands
    dist.run_commands()
  File "/Users/mdiener/gtk/inst/lib/python3.11/site-packages/setuptools/_distutils/dist.py", line 969, in run_commands
    self.run_command(cmd)
  File "/Users/mdiener/gtk/inst/lib/python3.11/site-packages/setuptools/dist.py", line 989, in run_command
    super().run_command(command)
  File "/Users/mdiener/gtk/inst/lib/python3.11/site-packages/setuptools/_distutils/dist.py", line 988, in run_command
    cmd_obj.run()
  File "/Users/mdiener/gtk/inst/lib/python3.11/site-packages/py2app/build_app.py", line 984, in run
    self._run()
  File "/Users/mdiener/gtk/inst/lib/python3.11/site-packages/py2app/build_app.py", line 1214, in _run
    self.run_normal()
  File "/Users/mdiener/gtk/inst/lib/python3.11/site-packages/py2app/build_app.py", line 1311, in run_normal
    self.process_recipes(mf, filters, flatpackages, loader_files)
  File "/Users/mdiener/gtk/inst/lib/python3.11/site-packages/py2app/build_app.py", line 1160, in process_recipes
    rval = check(self, mf)
           ^^^^^^^^^^^^^^^
  File "/Users/mdiener/gtk/inst/lib/python3.11/site-packages/py2app/recipes/sslmod.py", line 26, in check
    import ssl
  File "/Users/mdiener/gtk/inst/lib/python3.11/ssl.py", line 100, in <module>
    import _ssl             # if we can't import it, let the error propagate
    ^^^^^^^^^^^
ModuleNotFoundError: No module named '_ssl'
```

Reply:
Humm.. Will probably need py2app to skip the ssl module. 
This can probably be done here: https://github.com/yousseb/meld/blob/master/setup_py2app.py#L119

Reply:
Yeah, that *seems* to work, although it is a bit hard to tell cause starting from a fresh environment is difficult...